core/domain/takeout_service_test.py

```diff
@@ -876,6 +929,117 @@ def test_export_data_for_full_user_trivial_is_correct(self):
+            export_policy = model.get_export_policy()
+            renamed_export_keys = model.get_field_names_for_takeout()
+            exported_property_names = []
+            model_takeout_ids = []
+            for property_name in model._properties: # pylint: disable=protected-access
+                if (export_policy[property_name] ==
+                        base_models.EXPORT_POLICY.EXPORTED):
+                    if property_name in renamed_export_keys:
+                        exported_property_names.append(
+                            renamed_export_keys[property_name]
+                        )
+                    else:
+                        exported_property_names.append(property_name)
+                elif (export_policy[property_name] ==
+                      base_models.EXPORT_POLICY.EXPORTED_AS_MODEL_TAKEOUT_ID):
+                    model_takeout_ids.append(property_name)
+
+            if (export_method ==
+                    base_models
+                    .MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER):
+                self.assertEqual(len(exported_property_names), 0)
+                self.assertEqual(len(model_takeout_ids), 0)
+            elif (export_method ==
+                  base_models.MODEL_ASSOCIATION_TO_USER.ONE_INSTANCE_PER_USER):
+                exported_data = model.export_data(self.USER_ID_1)
+                self.assertEqual(len(model_takeout_ids), 0)
+                self.assertEqual(
+                    sorted([
+                        python_utils.UNICODE(key)
+                        for key in exported_data.keys()]),
+                    sorted(exported_property_names)
+                )
+            elif (export_method ==
+                  base_models
+                  .MODEL_ASSOCIATION_TO_USER
+                  .ONE_INSTANCE_SHARED_ACROSS_USERS):
+                self.assertIsNotNone(
+                    model.get_field_name_mapping_to_takeout_keys)
+                exported_data = model.export_data(self.USER_ID_1)
+                field_mapping = model.get_field_name_mapping_to_takeout_keys()
+                self.assertEqual(len(model_takeout_ids), 0)
+                self.assertEqual(
+                    sorted(exported_property_names),
+                    sorted(field_mapping.keys())
+                )
+                self.assertEqual(
+                    sorted(exported_data.keys()),
+                    sorted(field_mapping.values())
+                )
+            elif (export_method ==
+                  base_models
+                  .MODEL_ASSOCIATION_TO_USER.MULTIPLE_INSTANCES_PER_USER):
+                exported_data = model.export_data(self.USER_ID_1)
+                for model_id in exported_data.keys():
+                    if len(model_takeout_ids) > 0:
+                        self.assertEqual(
+                            model_id,
+                            getattr(model, model_takeout_ids[0])
```

Comment:
Why the [0]?

What if there is > 1 element?

All this probably needs more explanation / better naming.

Reply:
Done, let me know if still unclear.